### PR TITLE
[C++] [Qt5]Update isSet when the object is received from callback.

### PR DESCRIPTION
**Pull request by etherealjoy**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

`Changes:`
- set flag updated when deserializing from response, enabling object reuse
- forward declaration in template functions in helper class due to templates calling one another
- change name of `init` function due to symbol collision with generated code

Fixes: #4342
`cc:`
@muttleyxd @MartinDelille @stkrwork @ravinikam

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/cpp-qt5-client/helpers-header.mustache [hunk 1]
```diff
@@ -128,7 +149,7 @@ namespace {{this}} {
     bool fromJsonValue(QList<T> &val, const QJsonValue &jval) {
         bool ok = true;
         if(jval.isArray()){
-            for(const auto& jitem : jval.toArray()){
+            for(const auto jitem : jval.toArray()){
```
Comment (muttleyxd) [inline modules/openapi-generator/src/main/resources/cpp-qt5-client/helpers-header.mustache]: Is there a reason for removing reference here?

Comment (etherealjoy) [reply modules/openapi-generator/src/main/resources/cpp-qt5-client/helpers-header.mustache]: The compiler gives a warning that the method `QJsonValue .toArray()` cannot return a reference. So I removed it.

Comment (muttleyxd) [reply modules/openapi-generator/src/main/resources/cpp-qt5-client/helpers-header.mustache]: Looks good to me then!

**Comment by DevASG** [comment]
The same problem is related to the C++ Qt Server.
Is it possible to **generate Objects for Qt Server in the same way**?
Thanks.

**Comment by etherealjoy** [comment]
> The same problem is related to the C++ Qt Server.
> Is it possible to **generate Objects for Qt Server in the same way**?
> Thanks.

@DevASG I created #5634 for this.

**Pull request merged**